services/core-api/app/api/activity/models/activity_notification.py

```diff
@@ -161,7 +166,18 @@ def create_many(cls, mine_guid, activity_type, document, idempotency_key=None, c
             .with_entities(cls.notification_recipient) \
             .filter_by(idempotency_key=idempotency_key) \
             .all()] if idempotency_key else []
-        
+
+        if(renotify_period_minutes > 0):
+          # Calculate the datetime that was renotify_period_minutes minutes ago
+          renotify_timestamp = datetime.utcnow() - timedelta(minutes=renotify_period_minutes)
+          # Filter to retrived unread records for renotifying
+          unread_renotify_query = cls.query.with_entities(cls.notification_recipient)\
+                .filter_by(notification_read=False)\
+                .filter(cls.create_timestamp < renotify_timestamp)\
+                .order_by(cls.create_timestamp)
+
+        unread_renotify_users = [res[0] for res in unread_renotify_query.all()]
```

Comment:
I think this might produce an error if `renotify_period_minutes` is not set? As unread_renotify_query would be None

Reply:
Address this by defining an empty list for `unread_renotify_users` before the if block. variable name is updated as `unread_renotify_and_read_notify_users` because same list is used to filter the read/unread notifications to decide whether to send the notifications or not.